runtime/test/generator.ts

```diff
@@ -0,0 +1,188 @@
+import * as fc from 'fast-check';
+import * as reflection from '../src/reflection-type';
+import * as assert from 'assert';
+import * as _ from 'lodash';
+
+export class GenType extends fc.Arbitrary<any> {
+  private generator: fc.Arbitrary<any> | undefined;
+  constructor(private readonly type: reflection.Type, private readonly bias?: number) {
+    super();
+  }
+
+  withBias(freq: number): fc.Arbitrary<any> {
+    return freq === this.bias ? this : new GenType(this.type, freq);
+  }
+
+  generate(mrng: fc.Random): fc.Shrinkable<any> {
+    if (!this.generator) {
+      this.generator = this.makeGenerator();
+      if (this.bias !== undefined) {
+        this.generator = this.generator.withBias(this.bias);
+      }
+    }
+    return this.generator.generate(mrng);
+  }
+
+  private makeGenerator() {
+    const type = this.type;
+    if (type.type === 'string') {
+      if (type.enum) {
+        return fc.oneof(...type.enum.map(fc.constant));
+      }
+      return fc.hexaString();
+    }
+    if (type.type === 'integer') {
+      if (type.enum) {
+        return fc.oneof(...type.enum.map(fc.constant));
+      }
+      return fc.integer();
+    }
+    if (type.type === 'number') {
+      if (type.enum) {
+        return fc.oneof(...type.enum.map(fc.constant));
+      }
+      return fc.float().map(v => 1000 * v);
+    }
+    if (type.type === 'array') {
+      return fc.array(new GenType(type.items));
+    }
+    if (type.type === 'union') {
+      return fc.oneof(...type.options.map(option => new GenType(option)));
+    }
+    if (type.type === 'intersection') {
+      // todo: need to create all values and then merge them
+      assert.fail('todo intersection value generation');
+    }
+    if (type.type === 'named') {
+      return named(type.reference);
+    }
+    if (type.type === 'null') {
+      return fc.constant(null);
+    }
+    if (type.type === 'unknown') {
+      return fc.anything();
+    }
+    if (type.type === 'object') {
+      return new GenObject(type.properties, type.additionalProperties);
+    }
+    return assert.fail('todo arb for missing type ' + type);
+  }
+}
+
+class GenObject extends fc.Arbitrary<any> {
```

Comment:
I was thinking that this generator might be useful outside of tests... I at least have had to build oats types by hand before 😬 I believe there's one of those In `v8` still

Reply:
it cannot be in the distributed package because that would require fast-check which would just feel wrong. So somebody would need to take the time to create a new package. Hopefully in a new repo